h2/src/main/org/h2/mvstore/db/MVSortedTempResult.java

```diff
@@ -56,7 +56,7 @@
      * {@link #contains(Value[])} method is invoked. Only the root result should
      * have an index if required.
      */
-    private MVMap<ValueRow, Object> index;
+    private MVMap<ValueRow, ValueRow> index;
```

Comment:
`Object` / `true` were used here because `Boolean` uses only 1 byte. `ValueRow.getEmpty()` uses 2 bytes. `ValueNull.INSTANCE` or `ValueBoolean.TRUE` should use 1 byte too. The actual data type doesn't matter.

Reply:
Actually we need a `Set` here, but we have only a `Map`.

Reply:
Understood, I don't think that one byte would make a difference, but at this point I am not sure it was worth it at all to open that Pandora's box with generics, code becomes uglier, not cleaner. On top of all this my test show some 5-7% performance degradation, compare to the master, for no apparent reason.

Reply:
Here you can use `NULL` or `TRUE` instead of `ROW ()`, but it doesn't look like a possible source of slowdown, `MVSortedTempResult` is not used very often.

Reply:
No, of course not, it was just a general observation. 
Sure I can use TRUE, as it was, just requires more monkeying with map's generic types.

Reply:
You noticed that performance regression in MVStore on in the H2 with MVStore engine?

`master` branch has only a minor improvement for on-disk storages, an unnecessary wrapper is removed, but I don't think that it can cause such difference too.

Reply:
Performance regression is either gone (still not sure what it was) or offset by later improvements.